src/HealthChecks.Azure.Storage.Blobs/AzureBlobStorageHealthCheck.cs

```diff
@@ -31,21 +31,26 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
     {
         try
         {
-            // Note: BlobServiceClient.GetPropertiesAsync() cannot be used with only the role assignment
-            // "Storage Blob Data Contributor," so BlobServiceClient.GetBlobContainersAsync() is used instead to probe service health.
-            // However, BlobContainerClient.GetPropertiesAsync() does have sufficient permissions.
-            await _blobServiceClient
-                .GetBlobContainersAsync(cancellationToken: cancellationToken)
-                .AsPages(pageSizeHint: 1)
-                .GetAsyncEnumerator(cancellationToken)
-                .MoveNextAsync()
-                .ConfigureAwait(false);
-
             if (!string.IsNullOrEmpty(_options.ContainerName))
             {
+                // Note: PoLP (Principle of least privilege)
+                // This can be used having at least the role assignment "Storage Blob Data Reader" at container level or at least "Storage Blob Data Reader" at storage account level.
+                // See <see href="https://docs.microsoft.com/en-us/azure/storage/common/storage-auth-aad-app?tabs=dotnet#configure-permissions-for-access-to-blob-and-queue-data">Configure permissions for access to blob and queue data</see>
```

Comment:
nit: the XML syntax should be used only for `///` comments

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags

```suggestion
                // See https://docs.microsoft.com/en-us/azure/storage/common/storage-auth-aad-app?tabs=dotnet#configure-permissions-for-access-to-blob-and-queue-data
```